src/components/InputWithHightlights.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <div>
+    <div class="selected__container" :style="inputStyle">
+      <div
+        v-for="(part, index) in parts"
+        :key="index" class="selected__line"
+        :style="{ marginLeft: -inputScrolledLeft + 'px' }"
+      >
+        <span :style="{whiteSpace: 'pre'}">{{ part.textBefore }}</span>
+
+        <span
+          :class="['selected', `selected--color-${part.color}`]"
+          :style="{
+            whiteSpace: 'pre',
+          }"
+        >{{ part.text }}</span>
+      </div>
+    </div>
+
+    <unnnic-input
+      ref="input"
+      :value="value"
+      @input="$emit('input', $event)"
+      v-bind="$attrs"
+    />
+  </div>
```

Comment:
Replace with elements with semantics here. You can use the documentation on the Wiki as a basis